plugins/techdocs-backend/src/search/DefaultTechDocsCollator.ts

```diff
@@ -55,46 +60,43 @@ type EntityInfo = {
   kind: string;
 };
 
+/**
+ * A search collator responsible for gather and transform TechDocs document to index.
+ *
+ * @public
+ */
 export class DefaultTechDocsCollator implements DocumentCollator {
-  protected discovery: PluginEndpointDiscovery;
-  protected locationTemplate: string;
-  private readonly logger: Logger;
-  private readonly catalogClient: CatalogApi;
-  private readonly tokenManager: TokenManager;
-  private readonly parallelismLimit: number;
-  private readonly legacyPathCasing: boolean;
   public readonly type: string = 'techdocs';
   public readonly visibilityPermission = catalogEntityReadPermission;
 
-  /**
-   * @deprecated use static fromConfig method instead.
-   */
-  constructor(options: TechDocsCollatorOptions) {
```

Comment:
A `minor` changeset for this change, eh?  With instructions on how to update existing backend code.